### PR TITLE
Import standard cmake modules, standardise dependency resolution [ESD-1246] [ESD-1247]

### PR DESCRIPTION
**Pull request by woodfell**
Integrates with swift-nav/cmake#2

Import common cmake repository. Set up package cmake options to enable/disable unit tests. This repository does not have any dependencies.

### CODE REVIEW

File: c/test/CMakeLists.txt [hunk 1]
```diff
@@ -1,19 +1,11 @@
...
-    )
-    target_compile_options(rtcm_test PRIVATE "-std=gnu99")
-
-endif(CMAKE_CROSSCOMPILING)
+add_custom_command(
+    TARGET rtcm_test POST_BUILD
+    COMMENT "Running unit tests"
+    COMMAND rtcm_test
+)
+target_compile_options(rtcm_test PRIVATE "-std=gnu99")
```
Comment (jbangelo) [inline c/test/CMakeLists.txt]: Could we change this to set the `C_STANDARD` property instead? I guess that would also require bumping the minimum CMake version.

File: c/test/CMakeLists.txt [hunk 2]
```diff
@@ -1,19 +1,11 @@
-cmake_minimum_required(VERSION 2.8.7)
+add_executable (rtcm_test rtcm_decoder_tests.c)
 
-if (CMAKE_CROSSCOMPILING)
-    message(STATUS "Skipping unit tests, cross compiling")
+target_link_libraries(rtcm_test LINK_PUBLIC rtcm)
```
Comment (jbangelo) [inline c/test/CMakeLists.txt]: I think we should prefer using `PUBLIC` instead of `LINK_PUBLIC`.

File: c/test/CMakeLists.txt (hunk 1 above)
Comment (woodfell) [reply c/test/CMakeLists.txt]: Done

File: c/test/CMakeLists.txt (hunk 2 above)
Comment (woodfell) [reply c/test/CMakeLists.txt]: Removed it entirely, rtcm_test is an executable

**Pull request merged**